cmd/ui/src/views/Explore/ContextMenu/AssetGroupMenuItemV2.tsx

```diff
@@ -0,0 +1,163 @@
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+import { Button } from '@bloodhoundenterprise/doodleui';
+import { Dialog, DialogActions, DialogContent, DialogTitle, MenuItem } from '@mui/material';
+import { NodeResponse, apiClient, useExploreSelectedItem, useNotifications } from 'bh-shared-ui';
+import { FC, useState } from 'react';
+import { useMutation, useQuery } from 'react-query';
+import { selectOwnedAssetGroupId, selectTierZeroAssetGroupId } from 'src/ducks/assetgroups/reducer';
+import { toggleOwnedObjectNode, toggleTierZeroNode } from 'src/ducks/explore/actions';
+import { useAppDispatch, useAppSelector } from 'src/store';
+
+const AssetGroupMenuItem: FC<{ assetGroupId: number; assetGroupName: string }> = ({ assetGroupId, assetGroupName }) => {
+    const { addNotification } = useNotifications();
+    const dispatch = useAppDispatch();
+
+    const [open, setOpen] = useState(false);
+
+    const { selectedItemQuery } = useExploreSelectedItem();
+    const tierZeroAssetGroupId = useAppSelector(selectTierZeroAssetGroupId);
+    const ownedObjectAssetGroupId = useAppSelector(selectOwnedAssetGroupId);
+
+    const isMenuItemForTierZero = assetGroupId === tierZeroAssetGroupId;
+    const isMenuItemForOwnedObject = assetGroupId === ownedObjectAssetGroupId;
+
+    const mutation = useMutation({
+        mutationFn: ({ nodeId, action }: { nodeId: string; action: 'add' | 'remove' }) => {
+            return apiClient.updateAssetGroupSelector(assetGroupId, [
+                {
+                    selector_name: nodeId,
+                    sid: nodeId,
+                    action,
+                },
+            ]);
+        },
+        onSuccess: () => {
+            if (selectedItemQuery.data?.id) {
+                if (isMenuItemForTierZero) {
+                    dispatch(toggleTierZeroNode(selectedItemQuery.data.id));
+                } else if (isMenuItemForOwnedObject) {
+                    dispatch(toggleOwnedObjectNode(selectedItemQuery.data.id));
+                }
+            }
+
+            addNotification('Update successful.', 'AssetGroupUpdateSuccess');
+        },
+        onError: (error: any) => {
+            console.error(error);
+            addNotification('Unknown error, group was not updated', 'AssetGroupUpdateError');
+        },
+    });
+
+    const { data: assetGroupMembers } = useQuery(['listAssetGroupMembers', assetGroupId], () =>
+        apiClient
+            .listAssetGroupMembers(assetGroupId, undefined, {
+                params: {
+                    object_id: `object_id=eq:${(selectedItemQuery.data as NodeResponse)?.objectId}`,
+                },
+            })
+            .then((res) => res.data.data?.members)
+    );
+
+    const handleAddToAssetGroup = () => {
+        if (selectedItemQuery.data) {
+            mutation.mutate({ nodeId: (selectedItemQuery.data as NodeResponse).objectId, action: 'add' });
+        }
```

Comment:
Casting here seems pretty safe because this option is only available for nodes! As an option to avoid casting `as NodeResponse` could be able to check for the `objectId in selectedItem.data` before accessing it:

```suggestion
        if (selectedItemQuery.data && 'objectId' in selectedItemQuery.data) {
            mutation.mutate({ nodeId: selectedItemQuery.data.objectId, action: 'add' });
        }
```

Reply:
Updated! Great suggestion imo, and something I want to save to my toolbox for type checking, thanks again!

Reply:
There is also an exported type guard to do just this - see `isNode` from `useGraphItem`